hedera-node/src/main/java/com/hedera/services/ledger/accounts/BackingNfts.java

```diff
@@ -27,13 +27,16 @@
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
+import java.util.HashSet;
 import java.util.Set;
 import java.util.function.Supplier;
 
 import static com.hedera.services.utils.EntityNumPair.fromNftId;
 
 @Singleton
 public class BackingNfts implements BackingStore<NftId, MerkleUniqueToken> {
+	Set<NftId> existingNfts = new HashSet<>();
```

Comment:
Iirc there was a time we were so desperate to speed up `handleTransaction()` processing time that since the `FCMap.containsKey()` method is slower than `HashSet.contains()`, this redundant data structure made sense. 

But I think we now have enough performance breathing room that the memory cost of this extra data structure isn't worth it. (Keep in mind we test with 50M NFTs!)